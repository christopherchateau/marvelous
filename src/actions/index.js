export const storeCharacter = (character) => ({
  type: 'STORE_CHARACTER',
  character
})

export const updateStorageDetails = (currentIndex, length) => ({
  });
