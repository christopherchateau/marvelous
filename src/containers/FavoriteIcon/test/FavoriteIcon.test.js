import React from "react";
import { shallow } from "enzyme";
import { toggleFavorite } from "../../../actions";
import { FavoriteIcon, mapDispatchToProps } from "../index";

describe("FavoriteIcon", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <FavoriteIcon dispatchToggleFavorite={jest.fn()} id={1} />
    );
  });

  it("should exist", () => {
    expect(wrapper).toBeDefined();
  });

  it("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe("handleFavoriteClick", () => {
    it("should call handleFavoriteClick favorite icon is clicked", () => {
      const spy = spyOn(wrapper.instance(), "handleFavoriteClick");
      const mockEvent = { preventDefault: jest.fn() };
      wrapper.instance().forceUpdate();

      wrapper.find(".fa-star").simulate("click", mockEvent);
      expect(spy).toHaveBeenCalled();
    });
  });

  describe("mapDispatchToProps", () => {
    it("should dispatch toggleFavorite when dispatchToggleFavorite is called", () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = toggleFavorite(111);
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.dispatchToggleFavorite(111);
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });
  });
});
