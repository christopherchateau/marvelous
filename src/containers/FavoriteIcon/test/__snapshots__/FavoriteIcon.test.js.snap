// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoriteIcon should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FavoriteIcon
    dispatchToggleFavorite={[MockFunction]}
    favorited={true}
    id={1}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "fa-star fas favorited",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": "i",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "fa-star fas favorited",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": "i",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
